app/lib/signup/account_manager.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Signup
+  class AccountManager
+    def initialize(manager_account)
+      @manager_account = manager_account
+    end
+
+    attr_reader :manager_account
+
+    def create(signup_params, signup_result_class = ::Signup::Result)
+      transaction do
+        signup_result = build_signup_result(signup_params, signup_result_class)
+        yield(signup_result) if block_given?
+        save_result_with_plans(signup_result, signup_params) if signup_result.valid?
+        signup_result
+      end
+    end
+
+    private
+
+    delegate :transaction, to: :manager_account
+
+    class_attribute :account_builder, default: proc {}
+
+    # This method smells of :reek:TooManyStatements and :reek:DuplicateMethodCall
```

Comment:
This still smells this?

Reply:
Yup :smile: 
`:reek:DuplicateMethodCall` for 2 times `raise ActiveRecord::Rollback`, and `:reek:TooManyStatements` because it expects a method to have less lines.